Code/GraphMol/MolStandardize/MolStandardize.cpp

```diff
@@ -155,14 +155,12 @@ std::vector<std::string> enumerateTautomerSmiles(
   cleanup(*mol, params);
   MolOps::sanitizeMol(*mol);
 
-  auto *tautparams = new TautomerCatalogParams(params.tautomerTransforms);
-  //	unsigned int ntautomers = tautparams->getNumTautomers();
-  TautomerEnumerator te(new TautomerCatalog(tautparams));
+  TautomerEnumerator te(params);
 
-  std::vector<ROMOL_SPTR> res = te.enumerate(*mol);
+  auto res = te.enumerate(*mol);
```

Comment:
This looks to be a breaking change. I know it will affect some of my code that uses the Tautomer enumerator.

Reply:
It is, and I really don't like it (I have already been yelled at by @greglandrum ;-) ), but adding another optional `TautomerEnumeratorStatus*` parameter to `TautomerEnumerator::enumerate()` is not a very exciting alternative either. I had thought of making `TautomerEnumeratorResult` iterable; as you support this idea too I'll give it a go tomorrow then we may decide what we dislike less. Thanks for reviewing!

Reply:
I'm not fond of the backwards compatibility break either. My argument for why I think it's worth doing anyway is primarily that returning the results structures really is a lot cleaner and the enumeration code has only been out there for one release, so there hopefully won't be too much code that actually breaks.
Still, it's probably not that hard to shift me back to a: "let's not break backwards compatibility" position if you think it's important @bp-kelley

Reply:
I am working on a solution that I like much better: I hope you guys will like it too  :-)

Reply:
The `TautomerEnumeratorResult` structure is definitely an improvement.  Now it is iterable it should at least not break code that uses auto.

One way to preserve backwards compatibility would be to name the rewritten `enumerate` method something like `enumerateToResult` then have

```cpp
std::vector<ROMOL_SPTR> TautomerEnumerator::enumerate(const ROMol &mol) {
  auto result = enumerateToResult(mol);
  return result.tautomers();
}
```

Reply:
@jones-gareth That's certainly doable, even though it makes it less discoverable. @greglandrum Are you happy for me to go ahead with this change?

Reply:
@greglandrum @bp-kelley What about Gareth's suggestion to call the new method `enumerateToResult()` and leave the old `enumerate()` method returning a `ROMOL_SPTR` to avoid breaking existing code?
It is worth noting that to fix existing broken C++ builds it is sufficient to replace `ROMOL_SPTR` with `auto` or replace `enumerate()` with `enumerate()()`/`enumerate().tautomes()`. I could add a `#pragma` in the code to make it more obvious when looking at failed CI logs.
Waiting for your opinion here. Thanks in advance!

Reply:
@greglandrum @bp-kelley OK, Greg provided a good solution to the problem. I am finalizing this PR now and this should be ready for approval soon.